Meadowlark-js/packages/meadowlark-core/src/middleware/FindMetaEdModelMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { Logger, writeDebugStatusToLog, writeRequestToLog } from '../Logger';
+import { MiddlewareModel } from './MiddlewareModel';
+import { findMetaEdModel } from '../metaed/MetaEdModelFinder';
+
+const moduleName = 'FindMetaEdModelMiddleware';
+
+/**
+ * Finds the matching MetaEd model from the path components
+ */
+export async function metaeEdModelFinding({ frontendRequest, frontendResponse }: MiddlewareModel): Promise<MiddlewareModel> {
+  // if there is a response already posted, we are done
+  if (frontendResponse != null) return { frontendRequest, frontendResponse };
+  writeRequestToLog(moduleName, frontendRequest, 'metaeEdModelFinding');
+
+  const matchingMetaEdModel = await findMetaEdModel(frontendRequest.middleware.pathComponents);
+  if (matchingMetaEdModel == null) {
+    const errorMessage = 'FindMetaEdModelMiddleware.metaeEdModelFinding: Fatal error - matchingMetaEdModel not found';
+    Logger.error(errorMessage, frontendRequest.traceId);
+    const statusCode = 404;
```

Comment:
Is this a 404 or a 500? `ValidateResourceMiddleware.resourceValidation` rejects an invalid URL. So if somehow we don't have a matching metaed model here, then something has gone terribly wrong and it is not the "end user's" fault. That sounds like a 500.